internal/bminventory/inventory.go

```diff
@@ -4089,10 +4136,14 @@ func (b *bareMetalInventory) UpdateInfraEnvInternal(ctx context.Context, params
 				errWrapperLog = log.WithError(err)
 			}
 			errWrapperLog.Errorf("Failed to update InfraEnv with id %s", params.InfraEnvID)
+			tx.Rollback()
+			if r := recover(); r != nil {
+				tx.Rollback()
+			}
```

Comment:
Not sure if this is needed

Reply:
I think it is because you want to make sure you get the cluster update to include the usage and the actual infraEnv update or neither, right? (i.e. we don't want the infraEnv update to succeed but then fail and not get the usage update).